errors/errors.go

```diff
@@ -31,6 +32,11 @@ func NewInternalError(ctx context.Context, err error) InternalError {
 	return InternalError{err}
 }
 
+// NewInternalErrorFromString returns the custom defined error of type InternalError.
+func NewInternalErrorFromString(ctx context.Context, errorMessage string) InternalError {
```

Comment:
Method overloading :) could think about something like this http://changelog.ca/log/2015/01/30/golang but doesn't look very clean to me.

Reply:
It's not overloading actually. It's about allowing both string and error types in the param of this function. Alternative to this approach (two different functions) would be using interface{} as a param but not sure it's better because it's more error prone.